### PR TITLE
Resource renames

### PR DESCRIPTION
**Pull request by victorhurdugaci**
The names of the generated resources will change. This change addresses that. The new names are `<projectname>.<folder_path_with_dots>.<resource_name>`

Please review: @pranavkm @davidfowl @dougbu @DamianEdwards

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public class LinkTagHelper : TagHelper
         private const string FallbackTestClassAttributeName = "asp-fallback-test-class";
         private const string FallbackTestPropertyAttributeName = "asp-fallback-test-property";
         private const string FallbackTestValueAttributeName = "asp-fallback-test-value";
-        private const string FallbackJavaScriptResourceName = "compiler/resources/LinkTagHelper_FallbackJavaScript.js";
+        private const string FallbackJavaScriptResourceName = "Microsoft.AspNet.Mvc.TagHelpers.compiler.resources.LinkTagHelper_FallbackJavaScript.js";
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs]: Use namespace from type - don't hardcode it here.

**Comment by pranavkm** [comment]
:watch:

**Comment by dougbu** [comment]
@victorhurdugaci 
- which is the base change you're reacting to here?
- what is the timing for the set of changes?
- suggest this change should not impact the generated files under test/Microsoft.AspNet.Mvc.Razor.Host.Test/TestFiles.  instead change `MvcRazorHostTest.RunRuntimeTest()` and `...RunDesignTimeTest()` to use the same filenames as before (nput files are still read from disk) but check them against the renamed resources.

**Comment by dougbu** [comment]
:watch:

**Comment by victorhurdugaci** [comment]
@dougbu 

    which is the base change you're reacting to here?

https://github.com/aspnet/dnx/pull/1615

    what is the timing for the set of changes?

sometime this week
    
    suggest this change should not impact the generated files under ...

good point. updated

File: src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs [hunk 2]
```diff
@@ -40,7 +42,7 @@ public class LinkTagHelper : TagHelper
         private const string FallbackTestClassAttributeName = "asp-fallback-test-class";
         private const string FallbackTestPropertyAttributeName = "asp-fallback-test-property";
         private const string FallbackTestValueAttributeName = "asp-fallback-test-value";
-        private const string FallbackJavaScriptResourceName = "compiler/resources/LinkTagHelper_FallbackJavaScript.js";
+        private readonly string FallbackJavaScriptResourceName = Namespace + ".compiler.resources.LinkTagHelper_FallbackJavaScript.js";
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs]: nit: pls wrap this line

**Comment by dougbu** [comment]
:shipit:

**Pull request merged**